client/src/Components/CommonComponent/Sidebar/index.jsx

```diff
@@ -0,0 +1,259 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import { Icon } from 'antd';
+
+import './style.css';
+
+export default class Sidebar extends Component {
+  state = {
+    accounts: 'hidden',
+    reports: 'hidden',
+    msg: 'hidden',
+    settings: 'hidden',
+    arrow_down_accounts:'inline',
+    arrow_up_accounts:'hidden',
+    arrow_down_reports:'inline',
+    arrow_up_reports:'hidden',
+    arrow_down_msg:'inline',
+    arrow_up_msg:'hidden',
+    arrow_down_settings:'inline',
+    arrow_up_settings:'hidden',
+  }
+  handleClick = (value) => (e) => {
+    this.setState((prev) => {
+      let display = '';
+      let up='';
+      let down='';
+      if (prev[value] === 'hidden') {
+        display = 'block';
+        up='inline';
+        down='hidden';
+      } else {
+        display = 'hidden';
+        up='hidden';
+        down='inline';
+      }
+      if(value==='accounts')
+      return { [value]: display , arrow_down_accounts:down,arrow_up_accounts:up};
+      if(value==='reports')
+      return { [value]: display , arrow_down_reports:down,arrow_up_reports:up};
+      if(value==='msg')
+      return { [value]: display , arrow_down_msg:down,arrow_up_msg:up};
+      if(value==='settings')
+      return { [value]: display , arrow_down_settings:down,arrow_up_settings:up};
```

Comment:
can write it in a short way by :
- in `state` make it like that :
```
state = {
    accounts: {
      display: 'hidden',
      arrow: 'down'
    },
    accounts: {
      display: 'hidden',
      arrow: 'down'
    },
    reports: {
      display: 'hidden',
      arrow: 'down'
    },
    msg: {
      display: 'hidden',
      arrow: 'down'
    },
    settings: {
      display: 'hidden',
      arrow: 'down'
    },
  };
```
- for `handleClick` function the `value` represent the key of state like `accounts, msg ..`
for it can make all above code like that :
```
this.setState(prev => {
      console.log(value);
      console.log(prev);
      let display = ''; 
      let arrow = ''
      if(prev[value].display == 'hidden'){
        display = 'block';
        arrow = 'up'
      } else {
        display = 'hidden';
        arrow = 'down'
      }
      return {[value]: { display, arrow, }}
    });
```
- then just need in `render` put the value of state like that : 
```
 <Icon  type={accounts.arrow} />
<div className={`sidebar__dropdowncontainer ${accounts.display}`}>
```

Reply:
but I need to pass two different value to arrow 'hidden' and 'inline' for each arrow
and about the objects in the state we can use this style when I have another sidebar.